bin/morgue.js

```diff
@@ -3632,6 +3632,194 @@ function samplingUsage(str) {
+  if (attributes === undefined) {
+    attributes = [];
+  }
+  if (!Array.isArray(attributes)) {
+    attributes = [ attributes ];
+  }
+
+  let backoffsUnparsed = argv.backoff;
+  if (backoffsUnparsed === undefined) {
+    errx("At least one --backoff is required");
+  }
+  if (!Array.isArray(backoffsUnparsed)) {
+    backoffsUnparsed = [ backoffsUnparsed ];
+  }
+
+  /*
+   * Each backoff is count,interval.
+   */
+  let backoffs = [];
+  for (let unparsed of backoffsUnparsed) {
+    let split = unparsed.split(",");
+    if (split.length !== 2) {
+      errx("Usage of --backoff is --backoff count,interval");
+    }
+    let [countStr, intervalStr] = split;
+    let interval = parseTimeInt(intervalStr);
+    let count = Number.parseInt(countStr);
+    if (Number.isNaN(count)) {
+      errx("Backoff counts must be valid integers");
+    }
+    backoffs.push({ count, interval });
+  }
+
+  /* Set the non-optional fields. */
+  let config = {
+    type,
+    backoffs,
+    object_attributes: attributes,
+  };
+
+  if (argv["reset-interval"] !== undefined) {
+    if (Array.isArray(argv["reset-interval"])) {
+      errx("Only one --reset-interval is allowed");
+    }
+    config.reset_interval = parseTimeInt(argv["reset-interval"]);
+  }
+
+  config.missing_symbols = {};
+  const keepWhitelisted = argv["process-whitelisted"];
+  if (keepWhitelisted !== undefined) {
+    config.missing_symbols.process_whitelisted = parseBool(keepWhitelisted);
+  }
+  const keepPrivate = argv["process-private"];
+  if (keepPrivate !== undefined) {
+    config.missing_symbols.process_private = parseBool(keepPrivate);
+  }
+
+  let bucketsUnparsed = argv.buckets;
+  if (bucketsUnparsed !== undefined) {
+    if (typeof bucketsUnparsed !== 'number') {
+      errx("--buckets must be integer");
+    }
+    config.buckets = bucketsUnparsed;
+  }
+
+  const resetIntervalUnparsed = argv["reset-interval"];
+  if (resetIntervalUnparsed !== undefined) {
+    config.reset_interval = parseTimeInt(resetIntervalUnparsed);
+  }
+  return config;
+}
+
+function samplingConfigure(coroner, argv, config) {
+  let universe = argv.universe;
+  if (!universe) {
+    universe = Object.keys(config.config.universes)[0];
+  }
+
+  let project = argv.project;
+
+  if (!universe) {
+    errx("--universe is required");
+  }
+
+  if (!project) {
+    errx("--project is required");
+  }
+
+  let bpg = coronerBpgSetup(coroner, argv);
+  let model = bpg.get();
+
+  /* Find universe. */
+  let uid = 0;
+  for (let u of model.universe) {
+    if (u.get("name") === universe) {
+      uid = u.get("id");
+    }
+  }
+
+  if (uid == 0) {
+    errx("Universe not found");
+  }
+
+  let pid = 0;
+  for (let p of model.project) {
+    if (p.get("universe") === uid && p.get("name") == project) {
+      pid = p.get("pid");
+    }
+  }
+
+  if (pid === 0) {
+    errx("Project not found");
+  }
+  
+  let disabled = 0;
+  if (argv.disable) {
+    disabled = 1;
+  }
+
+  let projectSampling = null;
```

Comment:
That's the project's sampling configuration? The name is kind of ambiguous, mostly reads like a boolean than anything else.